### ISSUE TITLE
FAPI: policyRef comparison leads to tpm2-tools test suite failure

### ISSUE THREAD

**Issue by diabonas** [tpm2-software/tpm2-tss#1915]
With tpm2-tss master (bd08b5a874f260f0aa4bcfade93e1bb9a6e57a8c) or the latest release [3.0.2](https://github.com/tpm2-software/tpm2-tss/releases/tag/3.0.2), the tpm2-tools test suite fails both with the current master (https://github.com/tpm2-software/tpm2-tools/commit/684eeed0a133167a781f30c09c697142cd3861fc), which is the same as the latest release [5.0](https://github.com/tpm2-software/tpm2-tools/releases/tag/5.0).

The failing tests are [`fapi-branch-select` and `fapi-branch-select_ecc`](https://github.com/tpm2-software/tpm2-tools/blob/master/test/integration/fapi/fapi-branch-select.sh), the relevant part of the log looks like this:
```
FAIL: test/integration/fapi/fapi-branch-select
==============================================

[...]
run_startup: false
10+0 records in
10+0 records out
10485760 bytes (10 MB, 10 MiB) copied, 0,0611129 s, 172 MB/s
spawn tss2 sign --keyPath=HS/SRK/mySignKey --padding=RSA_PSS --digest=/tmp/tpm2_test_q6d5aO/tss2_fapi.zrBgH1/digest.file --signature=/tmp/tpm2_test_q6d5aO/tss2_fapi.zrBgH1/signature.file --publicKey=/tmp/tpm2_test_q6d5aO/tss2_fapi.zrBgH1/public_key.file
ERROR:fapi:src/tss2-fapi/ifapi_policy_callbacks.c:886:search_policy() ErrorCode (0x0006002e) Policy not found. 
ERROR:fapi:src/tss2-fapi/ifapi_policy_callbacks.c:1144:ifapi_exec_auth_policy() ErrorCode (0x0006002e) Search policy 
ERROR:fapi:src/tss2-fapi/ifapi_policy_execute.c:615:execute_policy_authorize() Execute authorized policy. ErrorCode (0x0006002e) 
ERROR:fapi:src/tss2-fapi/ifapi_policy_execute.c:1527:execute_policy_element() ErrorCode (0x0006002e) Execute policy authorize 
ERROR:fapi:src/tss2-fapi/ifapi_policy_execute.c:1737:ifapi_policyeval_execute() Execute policy. ErrorCode (0x0006002e) 
ERROR:fapi:src/tss2-fapi/ifapi_policyutil_execute.c:320:ifapi_policyutil_execute() Execute policy. ErrorCode (0x0006002e) 
WARNING:esys:src/tss2-esys/api/Esys_FlushContext.c:234:Esys_FlushContext_Finish() Received TPM Error 
ERROR:esys:src/tss2-esys/api/Esys_FlushContext.c:89:Esys_FlushContext() Esys Finish ErrorCode (0x000001c4) 
ERROR:fapi:src/tss2-fapi/fapi_util.c:2042:ifapi_authorize_object() Execute policy. ErrorCode (0x0006002e) 
ERROR:fapi:src/tss2-fapi/fapi_util.c:2737:ifapi_key_sign() ErrorCode (0x0006002e) Authorize signature key. 
ERROR:fapi:src/tss2-fapi/api/Fapi_Sign.c:299:Fapi_Sign_Finish() Fapi sign. ErrorCode (0x0006002e) 
ERROR:esys:src/tss2-esys/esys_iutil.c:1095:esys_GetResourceObject() Error: Esys handle does not exist (70018). 
ERROR:esys:src/tss2-esys/api/Esys_FlushContext.c:138:Esys_FlushContext_Async() flushHandle unknown. ErrorCode (0x00070018) 
ERROR:esys:src/tss2-esys/api/Esys_FlushContext.c:66:Esys_FlushContext() Error in async function ErrorCode (0x00070018) 
ERROR:fapi:src/tss2-fapi/api/Fapi_Sign.c:130:Fapi_Sign() ErrorCode (0x0006002e) Key_Sign 
Fapi_Sign(0x6002E) - fapi:The policy is unknown
[...]
FAIL test/integration/fapi/fapi-branch-select.sh (exit status: 1)
```

I have bisected this to commit https://github.com/tpm2-software/tpm2-tss/commit/419b48e190be8eb0ac24c8e084662c0420330167 / c4d7d24d72f0735c01c2088d3bd1be2c06b4a98b in the 3.0.x branch, so I suspect something might be off with the new policRef comparison:
```
commit c4d7d24d72f0735c01c2088d3bd1be2c06b4a98b (refs/bisect/bad)
Author: Juergen Repp <juergen.repp@sit.fraunhofer.de>
Date:   2020-10-24 12:45:57 +0200

    FAPI: Include policyRef into policy search for policy authorize.

    Currently only the public information of the signing key is used for policy search.
    Now also policyRef is included in the comparison.

    Signed-off-by: Juergen Repp <juergen.repp@sit.fraunhofer.de>

**Comment by JuergenReppSIT** [comment]
Yes the policyRef check is the reason. This problem should be solved with PR #2270.

**Comment by diabonas** [comment]
Indeed, I can confirm that applying https://github.com/tpm2-software/tpm2-tools/pull/2268 (for version 5.0/current master) makes the test suite pass, so I am closing this issue.
